### PR TITLE
dataloader: unit tests

### PR DESCRIPTION
**Pull request by Akasurde**
##### SUMMARY
Based on work from Alikins (https://github.com/ansible/ansible/pull/16500)

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
test/units/parsing/test_dataloader.py
test/units/parsing/vault.yml

### CODE REVIEW

**Comment by mattclay** [comment]
CI failure in Python 2.x unit tests: https://app.shippable.com/github/ansible/ansible/runs/102384/6/tests

**Comment by Akasurde** [comment]
@mattclay I updated the testcases. Now, they are passing.

File: test/units/parsing/vault.yml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+$ANSIBLE_VAULT;1.1;AES256
```
Comment (mattclay) [inline test/units/parsing/vault.yml]: This should probably be in `test/units/parsing/fixtures/` to match the other parsing unit tests.

Comment (Akasurde) [reply test/units/parsing/vault.yml]: Done.

**Comment by Akasurde** [comment]
@mattclay @bcoca Thanks for the review.

**Pull request merged**